### PR TITLE
Monitor

### PR DESCRIPTION
**Pull request by eile**

### CODE REVIEW

**Comment by eile** [comment]
FYI https://github.com/zeromq/libzmq/issues/2618

**Comment by eile** [comment]
Please re-review. Implementation needed polymorphism for pub and other sockets.

File: tests/common.h [hunk 1]
```diff
@@ -98,4 +98,23 @@ void onEchoEvent(const void* data, const size_t size)
...
+    }
+
+    Monitor(zeroeq::Sender& sender, zeroeq::Receiver& shared)
+        : zeroeq::Monitor(sender, shared)
+        , connections(0)
+    {
+    }
+
+    void notifyNewConnection() final { ++connections; }
+    size_t connections;
```
Comment (tribal-tec) [inline tests/common.h]: {0}

**Comment by tribal-tec** [comment]
Changelog and version(?)

**Pull request merged**